src/daemon.rs

```diff
@@ -92,68 +93,72 @@ impl Daemon {
         let (activity_tx, activity_rx) = chan::unbounded();
 
         let server = rpc::Server::new(socket_path, activity_tx, self.build_events_tx.clone())?;
-        let mut pool = crate::thread::Pool::new();
-        pool.spawn("accept-loop", move || {
-            server
-                .serve()
-                .expect("failed to serve daemon server endpoint");
-        })?;
         let build_events_rx = self.build_events_rx.clone();
         let mon_tx = self.mon_tx.clone();
-        pool.spawn("build-loop", move || {
-            let mut project_states: HashMap<NixFile, Event> = HashMap::new();
-            let mut event_listeners: Vec<chan::Sender<Event>> = Vec::new();
+        let mut pool = Pool::new(3);
+        pool.scoped(|scope| {
```

Comment:
Pretty much all the changes here are whitespace changes. I suggest hiding whitespace changes when reviewing.